editor/src/utils/canvas-react-utils.ts

```diff
@@ -211,37 +214,40 @@ const mangleExoticType = Utils.memoize(
   (type: React.ComponentType): React.FunctionComponent => {
     function updateChild(
       child: React.ReactElement,
-      dataUid: string | null,
+      dataUids: string | null,
       parentUid: string | null,
     ) {
-      const existingChildUID = child.props?.[UTOPIA_UID_KEY]
+      const existingChildUIDs = child.props?.[UTOPIA_UIDS_KEY]
+      const appendedUIDString = appendToUidString(existingChildUIDs, dataUids)
       const existingParentIDs = child.props?.[UTOPIA_UID_PARENTS_KEY]
       if ((!React.isValidElement(child) as any) || child == null) {
         return child
       } else {
         let pathParts: Array<string> = []
         // Added to here in reverse order, attempting to rebuild
         // what the path _should_ be.
-        if (typeof existingChildUID === 'string') {
-          pathParts.push(existingChildUID)
+        if (typeof existingChildUIDs === 'string') {
+          const childUIDArray = uidsFromString(existingChildUIDs)
+          pathParts.push(childUIDArray[0]) // TODO Balazs we are arbitrarily picking the first element here
         }
-        if (typeof dataUid === 'string') {
-          pathParts.push(dataUid)
+        if (typeof dataUids === 'string') {
+          const uidArray = uidsFromString(dataUids)
+          pathParts.push(uidArray[0]) // TODO Balazs we are arbitrarily picking the first element here
```

Comment:
Looks like this needs addressing. I don't mind pulling it into my spring cleaning work @balazsbajorics ?

Reply:
good catch – actually I think we should remove the pathParts altogether, alongside with `data-utopia-original-parents` – I don't mind if you deal with this in the `data-original-uid` PR